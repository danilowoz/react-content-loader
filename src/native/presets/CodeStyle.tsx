import * as React from 'react'

import { IContentLoaderProps } from '..'
import ContentLoader, { Rect } from '../ContentLoader'

const ReactContentLoaderCode: React.FC<IContentLoaderProps> = props => (
  <ContentLoader viewBox="0 0 340 84" width={340} height={84} {...props}>
    <Rect x="0" y="0" width="67" height="11" rx="3" />
    <Rect x="76" y="0" width="140" height="11" rx="3" />
    <Rect x="127" y="48" width="53" height="11" rx="3" />
    <Rect x="187" y="48" width="72" height="11" rx="3" />
    <Rect x="18" y="48" width="100" height="11" rx="3" />
    <Rect x="0" y="71" width="37" height="11" rx="3" />
    <Rect x="18" y="23" width="140" height="11" rx="3" />
    <Rect x="166" y="23" width="173" height="11" rx="3" />
  </ContentLoader>
)

export default ReactContentLoaderCode
