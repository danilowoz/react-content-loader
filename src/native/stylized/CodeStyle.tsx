import * as React from 'react'

import Holder, { Rect } from '../Holder'
import { IContentLoaderProps } from '../'

const ReactContentLoaderCode: React.FC<IContentLoaderProps> = props => (
  <Holder {...props}>
    <Rect x="0" y="0" rx="3" ry="3" width="70" height="10" />
    <Rect x="80" y="0" rx="3" ry="3" width="100" height="10" />
    <Rect x="190" y="0" rx="3" ry="3" width="10" height="10" />

    <Rect x="15" y="20" rx="3" ry="3" width="130" height="10" />
    <Rect x="155" y="20" rx="3" ry="3" width="130" height="10" />

    <Rect x="15" y="40" rx="3" ry="3" width="90" height="10" />
    <Rect x="115" y="40" rx="3" ry="3" width="60" height="10" />
    <Rect x="185" y="40" rx="3" ry="3" width="60" height="10" />

    <Rect x="0" y="60" rx="3" ry="3" width="30" height="10" />
  </Holder>
)

export default ReactContentLoaderCode
