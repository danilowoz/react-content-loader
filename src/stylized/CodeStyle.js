//@flow
import * as React from 'react'
import ContentLoader from '../index'
import type { WrapProps } from '../Wrap'

const CodeStyle = (props: WrapProps): React.Element<*> => (
  <ContentLoader {...props}>
    <rect x="0" y="0" rx="3" ry="3" width="70" height="10" />
    <rect x="80" y="0" rx="3" ry="3" width="100" height="10" />
    <rect x="190" y="0" rx="3" ry="3" width="10" height="10" />

    <rect x="15" y="20" rx="3" ry="3" width="130" height="10" />
    <rect x="155" y="20" rx="3" ry="3" width="130" height="10" />

    <rect x="15" y="40" rx="3" ry="3" width="90" height="10" />
    <rect x="115" y="40" rx="3" ry="3" width="60" height="10" />
    <rect x="185" y="40" rx="3" ry="3" width="60" height="10" />

    <rect x="0" y="60" rx="3" ry="3" width="30" height="10" />
  </ContentLoader>
)

export default CodeStyle
