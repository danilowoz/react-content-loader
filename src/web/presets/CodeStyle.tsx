import * as React from 'react'

import { IContentLoaderProps } from '..'
import ContentLoader from '../ContentLoader'

const ReactContentLoaderCode: React.FC<IContentLoaderProps> = props => (
  <ContentLoader viewBox="0 0 340 84" {...props}>
    <rect x="0" y="0" width="67" height="11" rx="3" />
    <rect x="76" y="0" width="140" height="11" rx="3" />
    <rect x="127" y="48" width="53" height="11" rx="3" />
    <rect x="187" y="48" width="72" height="11" rx="3" />
    <rect x="18" y="48" width="100" height="11" rx="3" />
    <rect x="0" y="71" width="37" height="11" rx="3" />
    <rect x="18" y="23" width="140" height="11" rx="3" />
    <rect x="166" y="23" width="173" height="11" rx="3" />
  </ContentLoader>
)

export default ReactContentLoaderCode
