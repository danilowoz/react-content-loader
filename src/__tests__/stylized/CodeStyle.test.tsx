import * as React from 'react'
import * as renderer from 'react-test-renderer'

import CodeStyle from '../../stylized/CodeStyle'

describe('CodeStyle', () => {
  const wrapper = renderer.create(<CodeStyle id="CodeStyle" speed={20} />)

  test('renders correctly', () => {
    const tree = wrapper.toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('props are propagated ', () => {
    expect(wrapper.root.props.speed).toEqual(20)
    expect(wrapper.root.props.id).toEqual('CodeStyle')
  })
})
