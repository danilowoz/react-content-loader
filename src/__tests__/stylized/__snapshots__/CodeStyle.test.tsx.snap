// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeStyle renders correctly 1`] = `
<svg
  aria-label="Loading interface..."
  preserveAspectRatio="none"
  role="img"
  style={Object {}}
  viewBox="0 0 400 130"
>
  <title>
    Loading interface...
  </title>
  <rect
    clipPath="url(#CodeStyle-idClip)"
    height={130}
    style={
      Object {
        "fill": "url(#CodeStyle-idGradient)",
      }
    }
    width={400}
    x="0"
    y="0"
  />
  <defs>
    <clipPath
      id="CodeStyle-idClip"
    >
      <rect
        height="10"
        rx="3"
        ry="3"
        width="70"
        x="0"
        y="0"
      />
      <rect
        height="10"
        rx="3"
        ry="3"
        width="100"
        x="80"
        y="0"
      />
      <rect
        height="10"
        rx="3"
        ry="3"
        width="10"
        x="190"
        y="0"
      />
      <rect
        height="10"
        rx="3"
        ry="3"
        width="130"
        x="15"
        y="20"
      />
      <rect
        height="10"
        rx="3"
        ry="3"
        width="130"
        x="155"
        y="20"
      />
      <rect
        height="10"
        rx="3"
        ry="3"
        width="90"
        x="15"
        y="40"
      />
      <rect
        height="10"
        rx="3"
        ry="3"
        width="60"
        x="115"
        y="40"
      />
      <rect
        height="10"
        rx="3"
        ry="3"
        width="60"
        x="185"
        y="40"
      />
      <rect
        height="10"
        rx="3"
        ry="3"
        width="30"
        x="0"
        y="60"
      />
    </clipPath>
    <linearGradient
      id="CodeStyle-idGradient"
    >
      <stop
        offset="0%"
        stopColor="#f0f0f0"
        stopOpacity={1}
      >
        <animate
          attributeName="offset"
          dur="20s"
          keyTimes="0; 0.25; 1"
          repeatCount="indefinite"
          values="-2; -2; 1"
        />
      </stop>
      <stop
        offset="50%"
        stopColor="#e0e0e0"
        stopOpacity={1}
      >
        <animate
          attributeName="offset"
          dur="20s"
          keyTimes="0; 0.25; 1"
          repeatCount="indefinite"
          values="-1; -1; 2"
        />
      </stop>
      <stop
        offset="100%"
        stopColor="#f0f0f0"
        stopOpacity={1}
      >
        <animate
          attributeName="offset"
          dur="20s"
          keyTimes="0; 0.25; 1"
          repeatCount="indefinite"
          values="0; 0; 3"
        />
      </stop>
    </linearGradient>
  </defs>
</svg>
`;
